test/y2storage/autoinst_proposal_test.rb

```diff
@@ -629,6 +629,101 @@
         expect { proposal.propose }.to raise_error(Y2Storage::UnexpectedCallError)
       end
     end
+
+    describe "when boot partition is required" do
+      let(:scenario) { "empty_hard_disk_50GiB" }
+
+      let(:boot_checker) do
+        instance_double(Y2Storage::BootRequirementsChecker, needed_partitions: [planned_boot])
+      end
+
+      let(:planned_boot) do
+        Y2Storage::Planned::Partition.new(nil).tap do |part|
+          part.min_size = 1.MiB
+          part.max_size = 1.MiB
+          part.partition_id = Y2Storage::PartitionId::BIOS_BOOT
+        end
+      end
+
+      before do
+        allow(Y2Storage::BootRequirementsChecker).to receive(:new).and_return(boot_checker)
+      end
+
+      let(:partitioning) do
+        [{ "use" => "all", "partitions" => [swap, root], "disklabel" => "gpt" }]
+      end
+
+      it "creates the boot partition" do
+        proposal.propose
+        devicegraph = proposal.devices
+        disk = devicegraph.disk_devices.first
+        expect(disk.name).to eq("/dev/sda")
+        expect(disk.partition_table.type).to eq(Y2Storage::PartitionTables::Type::GPT)
+      end
+    end
+
+    describe "partition table" do
```

Comment:
And we maybe should have some kind of test to ensure the partition table and the other partitions are already there when calling BootRequirementsChecker. That was part of the problem, anyway, that BootRequirementsChecker was called before creating the new partition tables.